src/erlcloud_ec2.erl

```diff
@@ -1441,27 +1538,131 @@ extract_block_device_mapping_status(Node) ->
+                              (aws_config()) -> ec2_normal_describe_return();
+                              (ec2_max_result()) -> ec2_paged_describe_return().
+describe_instance_status(InstanceID)
+    when is_list(InstanceID) ->
+    describe_instance_status([{"InstanceId", InstanceID}], [], default_config());
+describe_instance_status(Config)
+    when is_record(Config, aws_config) ->
+    describe_instance_status([], [], Config);
+describe_instance_status(MaxResults)
+    when is_integer(MaxResults) ->
+    describe_instance_status([], [], MaxResults, undefined, default_config()).
+
+-spec describe_instance_status(ec2_param_list(), filter_list()) -> ec2_normal_describe_return();
+                              (ec2_param_list(), aws_config()) -> ec2_normal_describe_return();
+                              (ec2_param_list(), ec2_max_result()) -> ec2_paged_describe_return();
+                              (ec2_max_result(), aws_config()) -> ec2_paged_describe_return();
+                              (ec2_max_result(), ec2_token()) -> ec2_paged_describe_return().
+describe_instance_status(Params, Filter)
+    when is_list(Params), is_list(Filter) ->
+    describe_instance_status(Params, Filter, default_config());
+describe_instance_status(Params, Config)
+    when is_list(Params), is_record(Config, aws_config) ->
+    describe_instance_status(Params, [], Config);
+describe_instance_status(Params, MaxResults)
+    when is_list(Params), is_integer(MaxResults) ->
+    describe_instance_status(Params, [], MaxResults, undefined, default_config());
+describe_instance_status(MaxResults, Config)
+    when is_integer(MaxResults), is_record(Config, aws_config) ->
+    describe_instance_status([], [], MaxResults, undefined, Config);
+describe_instance_status(MaxResults, NextToken)
+    when is_integer(MaxResults), is_list(NextToken) ->
+    describe_instance_status([], [], MaxResults, NextToken, default_config()).
+
+-spec describe_instance_status(ec2_param_list(), filter_list(), aws_config()) -> ec2_normal_describe_return();
+                              (ec2_param_list(), filter_list(), ec2_max_result()) -> ec2_paged_describe_return();
+                              (ec2_param_list(), ec2_max_result(), aws_config()) -> ec2_paged_describe_return();
+                              (ec2_param_list(), ec2_max_result(), ec2_token()) -> ec2_paged_describe_return();
+                              (ec2_max_result(), ec2_token(), aws_config()) -> ec2_paged_describe_return().
+describe_instance_status(Params, Filter, Config)
+    when is_list(Params), is_list(Filter), is_record(Config, aws_config) ->
     AllParams = Params ++ list_to_ec2_filter(Filter),
     case ec2_query(Config, "DescribeInstanceStatus", AllParams, ?NEW_API_VERSION) of
         {ok, Doc} ->
             Path = "/DescribeInstanceStatusResponse/instanceStatusSet/item",
             {ok, [ extract_instance_status(Item) || Item <- xmerl_xpath:string(Path, Doc) ]};
-        {error, _} = Error ->
```

Comment:
Why did you remove this line? Generally speaking it is more efficient. And readable (imo).